### PR TITLE
Ping: Stop requests if link is not open

### LINKED ISSUE #221: Ping sensor timers _periodicRequestTimer and _requestTimer continue to emit after catastrophic communication error
```
[13:21:58.882] ping.protocol.seriallink[Warning]: seriallink.cpp(26) Error is critical ! Port need to be closed.
[13:21:58.882] ping.protocol.seriallink[Warning]: seriallink.cpp(27) Error: QSerialPort::SerialPortError(ReadError)
[13:21:58.882] ping.protocol.seriallink[Debug]: seriallink.cpp(59) port closed
[13:21:59.288] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1213
[13:21:59.288] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1202
[13:21:59.288] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.297] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1213
[13:21:59.298] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1202
[13:21:59.349] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.416] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.483] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.550] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.605] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.669] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.697] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1213
[13:21:59.697] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1202
[13:21:59.733] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.797] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.861] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
[13:21:59.925] ping.protocol.ping[Debug]: ping.cpp(361) Requesting: 1300
```

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #221

This needs https://github.com/bluerobotics/ping-protocol-cpp/pull/48.

### CODE REVIEW

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -47,8 +47,18 @@ Ping::Ping() : Sensor()
...
             qCWarning(PING_PROTOCOL_PING) << "Can't write in this type of link.";
             _requestTimer.stop();
+            _periodicRequestTimer.stop();
             return;
         }
+
+        if(!link()->isOpen())
+        {
+            qCCritical(PING_PROTOCOL_PING) << "Can't write, port is not open!";
+            _requestTimer.stop();
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: Does this line belong here?

Looks like it should go (if not there already) in the _requestTimer callback. While it won't hurt here, it looks like a short piece of spaghetti.

Comment (patrickelectric) [reply src/sensor/ping.cpp]: This is the requestTimer callback, but I did what you said in requestPeriodicTimer

**Comment by patrickelectric** [comment]
@jaxxzer Done.

**Comment by patrickelectric** [comment]
Waiting CI to merge.

**Pull request merged**